tests/commands/global_test.py

```diff
@@ -147,3 +148,33 @@ def test_no_spawn_failure(self):
             refocus=False, on_success=self.on_success)
         cmd.apply(ui)
         ui.notify.assert_called_once_with('', priority='error')
+
+    @mock.patch('alot.commands.globals.settings.get', mock.Mock(return_value=''))
+    def test_spawn_no_stdin_success(self):
+        ui = mock.Mock()
+        with mock.patch.dict(os.environ, {'DISPLAY': ':0'}):
+            cmd = g_commands.ExternalCommand(
+                u'python -c "import sys; sys.exit(0)"',
+                refocus=False, spawn=True)
+        cmd.apply(ui)
+        ui.notify.assert_not_called()
+
+    @mock.patch('alot.commands.globals.settings.get', mock.Mock(return_value=''))
+    def test_spawn_stdin_success(self):
+        ui = mock.Mock()
+        with mock.patch.dict(os.environ, {'DISPLAY': ':0'}):
+            cmd = g_commands.ExternalCommand(
+                u'python -c "import sys; sys.exit(int(sys.stdin.read().strip()))"',
```

Comment:
`awk '{ exit $0 }'` takes 1/3 of python2, if you don't like the dependency on awk, `sh -c 'exit $(cat)'` takes 1/2 of python2.